### ISSUE TITLE
can.view.render() issues with rendering same template recursively

### ISSUE THREAD

**Issue by JeroGrav** [canjs/canjs#103]
This is only an issue when using can.view in cached mode.

If you are rendering something in a hierarchical manner - like a tree view for example. you normally render the same template recursively inside itself. This works fine with caching turned off but completely fails when caching is turned on.

---

Details:
say you have two templates: A and B and they look sort of like this

```
A: 
<div>
  ...
  <%==can.view.render("B", someData) %>
</div>

B:
<div>
  ...
  <%== can.view.render("A", someData.children) %>
</div>
```

the problem is that the second time you get to render A it will return undefined. The issue at hand is that the can.view.cache stores the deferred objects. So when the outer render of A happens, it gets the cached deferred and it acts fine. But when the inner render of A happens **it gets the same deferred as the outer render of A**.
This is a problem because the actual rendering is done in a deferred.then() call

```
snippet from can.jquery.js
    } else {
        // Otherwise, the deferred is complete, so
        // set response to the result of the rendering.
        deferred.then(function( renderer ) {
            response = renderer(data, helpers);
        });
    }

    return response;
}
```

The issue here is that the inner render of A needs to happen before the outer call to render A completes. However, the fact that it's a callback to deferred.then() means that the inner render of A doesn't happen until after the outer render of A has finished - which is too late.

As far as I can tell, there is no way around this as long as you cache the deferred. If you cache the renderer instead you won't have this problem.

Note: I've been testing this in synchronous mode but I don't believe it will work in async mode either.
